lib/change-layout.jsx

```diff
@@ -0,0 +1,61 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import {
+	showPane as showPaneAction,
+	hidePane as hidePaneAction,
+} from './state/settings/actions';
+import ThreePanesIcon from './icons/three-panes';
+import TwoPanesIcon from './icons/two-panes';
+import EditorOnlyIcon from './icons/editor-only';
+
+class ChangeLayout extends Component {
+
+	changeLayout = () => {
+		const {
+			showPane,
+			hidePane,
+			visiblePanes,
+		} = this.props;
+		if ( visiblePanes.includes( 'navigation-bar' ) ) {
+			hidePane( 'navigation-bar' );
+		} else if ( visiblePanes.includes( 'note-list' ) ) {
+			hidePane( 'note-list' );
+		} else {
+			showPane( 'note-list' );
+			showPane( 'navigation-bar' );
+		}
+	}
+
+	render() {
+		const { visiblePanes } = this.props;
+		return (
+			<button
+				title="Change Layout"
+				className="button button-borderless"
+				onClick={ this.changeLayout }
+			>
+				{ visiblePanes.includes( 'navigation-bar' ) &&
+					<TwoPanesIcon />
+				}
+				{ ! visiblePanes.includes( 'navigation-bar' ) && visiblePanes.includes( 'note-list' ) &&
+					<EditorOnlyIcon />
+				}
+				{ ! visiblePanes.includes( 'note-list' ) && ! visiblePanes.includes( 'navigation-bar' ) &&
+					<ThreePanesIcon />
+				}
+			</button>
```

Comment:
Are we showing a single icon and rotating the view based on repeated clicks to that icon? I'm not sure how I feel about this if it's true. Do I have to click twice to go back to the previous view?

I'm a big fan of how the Mac app handles the tag list. There's an arrow one can click on to open and close it - the drawer idiom.